### PR TITLE
Expand and collapse views programmatically

### PR DESCRIPTION
**Pull request by dgreenhalgh**
Finishes #55.

Views can now be expanded or collapsed programmatically. Added API calls to expand/collapsed keyed by `ParentObject` or by the index of the parent in the list of parents. Also added API calls to expand/collapse all parents.

Probably need to improve on some of the repetitive method names and could clean up some of the code in the `ExpandableRecyclerAdapter`. 

Note that I haven't added an API to expand/collapse a range of parents. That can come in a future PR.

### CODE REVIEW

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 1]
```diff
@@ -172,68 +175,192 @@ public int getItemViewType(int position) {
...
-        ParentWrapper parentWrapper = (ParentWrapper) getHelperItem(position);
+    public void expandParent(int parentIndex) {
+        int parentWrapperIndex = getParentWrapperIndex(parentIndex);
+
+        Object helperItem = getHelperItem(parentWrapperIndex);
+        ParentWrapper parentWrapper = null;
+        if (helperItem instanceof ParentWrapper) {
+             parentWrapper = (ParentWrapper) helperItem;
+        }
+        if (parentWrapper == null) {
```
Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Seems like this bit of if's can be cleaned up some, like this (can't see if parentwrapper is actually needed in this case)?
```java
if (helperItem instanceof ParentWrapper) {
    expandViews(parentWrapperIndex);
}
```

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 2]
```diff
@@ -342,4 +503,52 @@ public void onRestoreInstanceState(Bundle savedInstanceStateBundle) {
...
+     *         Otherwise, returns {@value null}.
+     */
+    private ParentWrapper getParentWrapper(ParentObject parentObject) {
+        ParentWrapper parentWrapper = null;
+        int numHelperItems = mHelperItemList.size();
+        for (int i = 0; i < numHelperItems; i++) {
+            Object helperItem = mHelperItemList.get(i);
+            if (helperItem instanceof ParentWrapper) {
+                if (((ParentWrapper) helperItem).getParentObject().equals(parentObject)) {
+                    parentWrapper = (ParentWrapper) helperItem;
```
Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Can we return at this point (and return null at the bottom if none matched)?

Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Good catch. Added `break` to match `getParentWrapperIndex(int)` above.

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 3]
```diff
@@ -172,68 +175,226 @@ public int getItemViewType(int position) {
...
+    }
+
+    /**
+     * Calls through to the {@link ParentViewHolder} to expand views for each
+     * {@link RecyclerView} a specified parent is a child of. These calls to
+     * the {@code ParentViewHolder} are made so that animations can be
+     * triggered at the {@link android.support.v7.widget.RecyclerView.ViewHolder}
+     * level.
+     *
+     * @param parentIndex The inject of the parent to expand
```
Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: typo: s/inject/index

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 4]
```diff
@@ -172,68 +175,226 @@ public int getItemViewType(int position) {
...
+    }
+
+    /**
+     * Calls through to the {@link ParentViewHolder} to collapse views for each
+     * {@link RecyclerView} a specified parent is a child of. These calls to
+     * the {@code ParentViewHolder} are made so that animations can be
+     * triggered at the {@link android.support.v7.widget.RecyclerView.ViewHolder}
+     * level.
+     *
+     * @param parentIndex The inject of the parent to collapse
```
Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: same typo

**Pull request merged**